stor/dx.py

```diff
@@ -204,6 +204,18 @@ def resource(self):
         joined_resource = '/'.join(parts[1:]) if len(parts) > 1 else None
         return self.parts_class(joined_resource) if joined_resource else None
 
+    def normpath(self):
+        colon_pieces = self[len(self.drive):].split(':', 1)
+        project = colon_pieces[0]
+        resource = (colon_pieces[1] if len(colon_pieces) == 2 else '').lstrip('/')
+        normed_resource = self.path_module.normpath('/' + resource)[1:]
+        norm_pth = self.path_class(self.drive + project + ':/' + normed_resource)
+        if isinstance(norm_pth, DXCanonicalPath):
```

Comment:
question:

why not just split out canonical path to a separate implementation for each type?  Then canonical path is just something like:

```
return self.path_class(self.drive + self.project + ':' + self.resource.lstrip('/'))
```

and for virtualpath you don't need to have that if statement anymore.

(i.e. `normed_resource = self.path_module.normpath(self.resource).lstrip('/')`, etc)

Reply:
I actually started with that, but ran into a subtle issue. `stor.Path('dx://project-123456789012345678901234:/a/../file-123456789012345678901234')` is initialized to a DXVirtualPath, so if we call normpath() on this, it'll implement the virtual version and return `DXVirtualPath("dx://project-123456789012345678901234:/file-123456789012345678901234")`, whereas we would expect `DXCanonicalPath("dx://project-123456789012345678901234:file-123456789012345678901234")`.

This helps prevent that. The main catch here is that `norm_pth = self.path_class(self.drive + project + ':/' + normed_resource)` determines the initialization class. Because `stor.dx.DXVirtualPath('dx://project-123456789012345678901234:/file-123456789012345678901234')` is initialized to `DXCanonicalPath("dx://project-123456789012345678901234:/file-123456789012345678901234")`.

Reply:
seems like this that should return a DXCanonicalPath and you have a good test case for it :)

(also that's just a really weird example)

Reply:
Addressed.